pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/AggregationFunctionFactory.java

```diff
@@ -112,8 +111,10 @@ public static AggregationFunction getAggregationFunction(AggregationInfo aggrega
           case DISTINCTCOUNTRAWHLLMV:
             return new DistinctCountRawHLLMVAggregationFunction();
           case DISTINCT:
+            Preconditions.checkState(brokerRequest != null,
+                "Broker request must be provided for 'DISTINCT' aggregation function");
             return new DistinctAggregationFunction(AggregationFunctionUtils.getColumn(aggregationInfo),
-                brokerRequest != null ? brokerRequest.getLimit() : SelectAstNode.DEFAULT_RECORD_LIMIT, brokerRequest.getOrderBy());
+                brokerRequest.getOrderBy(), brokerRequest.getLimit());
```

Comment:
I am unable to recall why I had this if check. I think there was a code-path where null broker request could have been passed. It is not possible right?

Reply:
The existing code never works for `null` brokerRequest because `brokerRequest.getOrderBy()` will directly throw NPE. The broker aggregation reducer might call this without brokerRequest, but distinct is not using that code path to reduce the result.
I added TODOs in `AggregationFunctionUtils` to always pass broker request for all code paths